### DISCUSSION TITLE
what's the difference between `load_from_checkpoint ` and `resume_from_checkpoint`

### DISCUSSION THREAD

**Discussion by marsggbo** [Lightning-AI/pytorch-lightning#11378] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I'm confused about two API:
- `Module.load_from_checkpoint`
- `trainer.resume_from_checkpoint`

**Comment by rohitgr7** [comment, marked as answer]
[resume_from_checkpoint](https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#resume-from-checkpoint) is used to resume the training using the checkpointed state_dicts. It will reload model's state_dict, optmizer's and schedulers's state_dicts, training state as well in a general case.
use-case: to restart the training

[load_from_checkpoint](https://pytorch-lightning.readthedocs.io/en/stable/common/weights_loading.html#checkpoint-loading) just reloads the model's state_dict and return the model with the loaded weights.
use-case: for quick evaluation/prediction.

**Reply by marsggbo** [reply]
thanks!
